onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -704,6 +713,32 @@ bool ReshapeOpSupportChecker::IsOpSupportedImpl(const InitializedTensorSet& init
   return true;
 }
 
+bool ReshapeOpSupportChecker::HasSupportedInputOutputsImpl(
+    const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+    const OpSupportCheckParams& params) const {
+  int32_t input_type;
+  if (!GetType(node_unit.Inputs()[0].node_arg, input_type))
+    return false;
+
+  if (input_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
```

Comment:
These are not required here, this is only for these operators
https://github.com/microsoft/onnxruntime/blob/bfb20b315d9763f4113fdb1b15444626d66874e6/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc#L286-L293